course02/project01/gavin/Asteroids/game_object.cc

```diff
@@ -0,0 +1,29 @@
+//
+// Created by gavinparker on 10-2-19.
+//
+
+#include <algorithm>
+
+#include "game_object.h"
+
+void GameObject::Update(float frameDelta)
+{
+    mCallbacks.erase(std::remove_if(mCallbacks.begin(), mCallbacks.end(), [this](auto &callback){return Call(callback);}), mCallbacks.end());
+    mChildren.erase(std::remove_if(mChildren.begin(), mChildren.end(), [](auto &child){return child->GetDestroyed();}), mChildren.end());
+    std::for_each(mChildren.begin(), mChildren.end(), [frameDelta](auto child){child->Update(frameDelta);});
+}
+
+bool GameObject::Call(Callback &callback)
```

Comment:
As I mentionned in my general comment this function should typically be part of an event loop, not an object

Reply:
This is part of the ```GameWorld``` in 76db252356e796ace0396de1cb5a68c9790e3d06